consensus/consensus-types/src/block.rs

```diff
@@ -60,6 +63,25 @@ impl Display for Block {
     }
 }
 
+#[cfg(any(test, feature = "fuzzing"))]
+impl Arbitrary for Block {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        any::<BlockData>()
+            .prop_map(|block_data| {
+                let id = block_data.hash();
+                Self {
+                    id,
+                    block_data,
+                    signature: None,
```

Comment:
this will not be very effective since it fails the check of proposal must have a signature, probably can have a oneof(Just(None), any::<Signature>())